src/EventLoop.hs

```diff
@@ -122,59 +122,57 @@ runGithubEventLoop ghQueue enqueueEvent = runLoop
       runLoop
 
 runLogicEventLoop
-  :: MonadIO m
-  => MonadLogger m
-  => MonadMonitor m
+  ::  forall es
+  .  IOE :> es
+  => Metrics.MetricsOperation :> es
+  => Time.TimeOperation :> es
+  => Git.GitOperation :> es
+  => GithubApi.GithubOperation :> es
+  => MonadLoggerEffect :> es
   => TriggerConfiguration
   -> ProjectConfiguration
   -> MergeWindowExemptionConfiguration
-  -- Interpreters for Git and GitHub actions.
-  -> (forall a. Metrics.MetricsOperationFree a -> m a)
-  -> (forall a. Time.TimeOperationFree a -> m a)
-  -> (forall a. Git.GitOperationFree a -> m a)
-  -> (forall a. GithubApi.GithubOperationFree a -> m a)
   -- Action that gets the next event from the queue.
-  -> m (Maybe Logic.Event)
+  -> IO (Maybe Logic.Event)
   -- Action to perform after the state has changed, such as
   -- persisting the new state, and making it available to the
   -- webinterface.
-  -> (ProjectState -> m ())
+  -> (ProjectState -> IO ())
   -> ProjectState
-  -> m ProjectState
+  -> Eff es ProjectState
 runLogicEventLoop
   triggerConfig projectConfig mergeWindowExemptionConfig
-  runMetrics runTime runGit runGithub
   getNextEvent publish initialState =
   let
     repo             = Config.repository projectConfig
-    runAll           = foldFree (runSum (runSum runMetrics runTime) (runSum runGit runGithub))
-    runAction        = foldFree (runSum (Logic.runBaseAction projectConfig) (Logic.runRetrieveEnvironment projectConfig))
```

Comment:
Here's a benefit of `effectful`: these two functions are rather awkward. They build and immediately destroy some sort of free monad tree. This is hard to add new effects to. In `effectful`, adding a new effect is less effort.

Later down in the PR you will see that the whole `Action` concept has disappeared in favor of just the `BaseAction` and `RetrieveEnvironment` effects.